### PR TITLE
Show branch name near commit message box

### LINKED ISSUE #73644: Show branch name near commit message box
Issue Type: <b>Feature Request</b>

It would be sweat to show the current branch name **near** the commit message box.
Many times I committed the changes to the wrong branch (I know we can see it in the status bar. I just forgot to take a look)

where

---

VS Code version: Code 1.33.1 (51b0b28134d51361cf996d2f0a1c698247aeabd8, 2019-04-11T08:27:14.102Z)
OS version: Windows_NT x64 10.0.18895

### PR DESCRIPTION
**Pull request by Devon7925**
resolves #73644

### CODE REVIEW

**Review by Drockk — APPROVED** [review]

File: extensions/git/src/repository.ts [hunk 1]
```diff
@@ -626,6 +626,10 @@ export class Repository implements Disposable {
 		this.status();
 	}
 
+	updateInputBox() {
+		this._sourceControl.inputBox.placeholder = 'Message ({0} to commit to ' + (this._HEAD ? this._HEAD.name || '' : '') + ')';
```
Comment (burknator) [inline extensions/git/src/repository.ts]: I'm pretty sure this breaks the localization of the string. See my comment above....

File: extensions/git/src/repository.ts [hunk 2]
```diff
@@ -577,7 +577,7 @@ export class Repository implements Disposable {
 
 		const root = Uri.file(repository.root);
 		this._sourceControl = scm.createSourceControl('git', 'Git', root);
-		this._sourceControl.inputBox.placeholder = localize('commitMessage', "Message (press {0} to commit)");
+		this._sourceControl.inputBox.placeholder = localize('commitMessage', "Message ({0} to commit)");
```
Comment (burknator) [inline extensions/git/src/repository.ts]: I think if you use `this.onDidRunGitStatus()` to set the placeholder whenever something in the repository changes would replace your `updateInputBox()` method (see my PR #75295).

File: extensions/git/src/repository.ts (hunk 1 above)
Comment (joaomoreno) [reply extensions/git/src/repository.ts]: @burknator Is right. Let's fix this.

Also, if `this._HEAD` doesn't exist, the message will become `Message ({0} to commit to )`. Let's fix that too.

File: extensions/git/src/repository.ts (hunk 2 above)
Comment (joaomoreno) [reply extensions/git/src/repository.ts]: Let's just call `updateInputBox()` here.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/repository.ts (hunk 1 above)
Comment (importnetminecraft) [reply extensions/git/src/repository.ts]: Fixed

File: extensions/git/src/repository.ts (hunk 2 above)
Comment (importnetminecraft) [reply extensions/git/src/repository.ts]: Done

File: extensions/git/src/repository.ts [hunk 3]
```diff
@@ -727,6 +727,12 @@ export class Repository implements Disposable {
 		this.updateCommitTemplate();
 	}
 
+	updateInputBox() {
+		const headless: Boolean = this.headLabel === '';
+		// '{0}' will be replaced by the corresponding key-command later in the process, which is why it needs to stay.
+		this._sourceControl.inputBox.placeholder = localize(headless ? 'commitMessage' : 'commitMessageWithHeadLabel', "Message ({0} to commit{1})", "{0}", headless ? 'this.headLabel' : ' to ' + this.headLabel);
```
Comment (joaomoreno) [inline extensions/git/src/repository.ts]: As @burknator [mentions](https://github.com/microsoft/vscode/issues/73644#issuecomment-500998333): this _breaks localization of the placeholder_.

**Comment by joaomoreno** [comment]
Will take https://github.com/microsoft/vscode/pull/75295 instead. Sorry about that. Thanks for your contribution any way! :pray:

**Pull request closed without merging**